internal/app/app.go

```diff
@@ -0,0 +1,94 @@
+package app
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Makpoc/gomigen/internal/generator"
+	"github.com/Makpoc/gomigen/internal/parser"
+	"github.com/Makpoc/gomigen/internal/version"
+)
+
+type Logger interface {
+	Printf(format string, v ...interface{})
+}
+
+type App struct {
+	pkgDir          string
+	interfaceName   string
+	outputDirectory string
+	version         string
+	logger          Logger
+}
+
+func New(pkgDir, interfaceName string, opts ...Option) *App {
+	app := &App{
+		pkgDir:        pkgDir,
+		interfaceName: interfaceName,
+		logger:        log.Default(),
+		version:       version.Version(),
+	}
+	for _, o := range opts {
+		o(app)
+	}
+	return app
+}
+
+func (a *App) Run() error {
+	a.logger.Printf("Looking for interface %q in %q", a.interfaceName, a.pkgDir)
```

Comment:
NIT: I personally don't think that so many log messages are needed. I think it is fine to have a single line that indicates that the generated middleware is being saved to a given file (so basically the last `Printf`). That way when I run the tool over a larger package, I would get one line per interface and it would be easier to visually process.

I personally don't think anyone cares about the intermediate state / process. If there is a problem along the way, that should be reported back through the error messages.

Reply:
I thought about adding a leveled logger to help with debugging issues in verbose mode, but for now I've removed all but the last log entry. I also moved the last log to only log on success.